live-examples/js-examples/statement-async.html

```diff
@@ -0,0 +1,21 @@
+<pre>
+<code id="static-js">function resolveAfter2Seconds(x) {
+  return new Promise(resolve => {
+    console.log("waiting...");
+    setTimeout(() => {
+      resolve(x);
+    }, 2000);
+  });
+}
+
+async function add1(x) {
+  var a = await resolveAfter2Seconds(20);
+  return(a + x);
+}
+
+add1(10).then(v => {
+  console.log(v);
+  // expected output: 30
+});
+</code>
+</pre>
```

Comment:
I might be missing something, but I thought something like this might be simpler:

```
function resolveAfter2Seconds(x) {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve(x);
    }, 2000);
  });
}

async function add1() {
  console.log("calling");
  var a = await resolveAfter2Seconds(20);
  console.log(a);
  console.log("finished");
}

add1();
```
It's still pretty long, but to me it illustrates more clearly the way you can use async/await to write linear async code. Or am I missing something?

Reply:
Sure! Simple is good! Although the example size will grow somewhat once we put the `// expected output:` comments in. I do wonder whether we need `// expected output:` comments for all `console.log`s.

Reply:
Yes, agreed: as we discussed:

* I don't think always we need `// expected output:`, especially in cases like this where the expected output is very obvious.

* we could omit one of the logs, by doing something like:

```
function resolveAfter2Seconds() {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve("finished");
    }, 2000);
  });
}

async function callAsync() {
  console.log("calling");
  var result = await resolveAfter2Seconds();
  console.log(result);
}

callAsync();
```